src/lib/device/round_robin_chunk_selector.h

```diff
@@ -43,14 +35,12 @@ class RoundRobinChunkSelector : public ChunkSelector {
     RoundRobinChunkSelector& operator=(RoundRobinChunkSelector&&) noexcept = delete;
     ~RoundRobinChunkSelector() = default;
 
-    void add_chunk(cshared< Chunk >& chunk) override;
-
-    Chunk* select(blk_count_t nblks, const blk_alloc_hints& hints) override;
-
-    void foreach_chunks(std::function< void(cshared< Chunk >&) >&& cb) override;
+    void add_chunk(VChunk chunk) override;
+    Chunk* select_chunk(blk_count_t nblks, const blk_alloc_hints& hints) override;
+    void foreach_chunks(std::function< void(VChunk&) >&& cb) override;
```

Comment:
Is there a way to avoid std::function, to auto& and make use of lambda. If not, we don't want to call foreach_chunks on very critical path. I know it is a micro optimization, but one that accumulates over time.

Reply:
it tried , but the compiler throws an error that we can not use `auto&` or `auto&&` in a virtual function declaration